pkg/rtc/participant.go

```diff
@@ -1234,6 +1235,9 @@ func (p *ParticipantImpl) onMediaTrack(track *webrtc.TrackRemote, rtpReceiver *w
 	}
 
 	publishedTrack, isNewTrack := p.mediaTrackReceived(track, rtpReceiver)
+	if isNewTrack {
+		prometheus.AddPublishAttempt(track.Kind().String())
+	}
```

Comment:
ok commenting after looking at these prom APIs. This should actually be publish success counter. At this point, we have received first packet(s) of the media stream, i. e. a successful publish.

Reply:
Thinking more about this, if you are using telemetry events for success counter (i. e. track published, track subscribed events for incrementing success counter), we should probably standardise that. And add attempt counter at appropriate place.

In that case, need to be careful about attempts as both remote and local tracks could increment attempt (the subscription part), but event would happen only once.

Or the supervisor could be one place where all this could happen

For publication:
- Adding publication attempt can happen [here](https://github.com/livekit/livekit/blob/b0a6bb3bc9c1ecbf9c3da1e93ac11cd3027ade9b/pkg/rtc/supervisor/publication_monitor.go#L69).
- Successful publication attempt can happen [here](https://github.com/livekit/livekit/blob/b0a6bb3bc9c1ecbf9c3da1e93ac11cd3027ade9b/pkg/rtc/supervisor/publication_monitor.go#L173). - If not returning there, add some code like `if pub.isStart { add_publication_success_counter }`

For subscription:
- Add subscription attempt [here](https://github.com/livekit/livekit/blob/b0a6bb3bc9c1ecbf9c3da1e93ac11cd3027ade9b/pkg/rtc/supervisor/subscription_monitor.go#L75) - I think you already have this
- Similar to publication ☝️ , can do the success counter update in `update()` function around [here](https://github.com/livekit/livekit/blob/b0a6bb3bc9c1ecbf9c3da1e93ac11cd3027ade9b/pkg/rtc/supervisor/subscription_monitor.go#L172). Note that subscription could happen due to a lot of reasons (auto subscribe, manually subscribing/unsubscribing, adaptive stream, permissions, moving between local and remote tracks, etc.). I think telemetry `TrackSubscribed` event should happen for all (except maybe for moving between remote and local and vice-versa). But, putting it all in supervisor could be a good way to capture it all in one central place.

On that topic, wondering if we should add counters for unsubscription attempts/successes.

Reply:
Moved everything into respective publication/subscription monitor to keep this uniform. For unsubscribe; I'm not sure how that counter would work, only logic around it I see is at https://github.com/livekit/livekit/blob/master/pkg/rtc/supervisor/subscription_monitor.go#L126-L130

Reply:
It would be similar to `subscribe` operation @real-danm . I left a new comment in the `subscribe` path. Adding on to that comment, the unsubscribe attempt count would be added if `if !IsSubscribe` and also in the `update()` function, the `unsubscribe` operation completion check is the second part. So, can increase the `success` counter based on second part.

BTW, don't need to to do this now. We can check the `subscribe` stuff first and check for any additional things we may need and decide on this later.